xonsh/ptk_shell/shell.py

```diff
@@ -219,8 +223,8 @@ def singleline(
             # once the prompt is done, update it in background as each future is completed
             prompt_args["pre_run"] = self.prompt_formatter.start_update
 
+        events.on_pre_prompter_prompt.fire()
```

Comment:
Doesn't this need to be added to the readline shell then too?

Reply:
We're thinking similar and I already take a look at readline when worked on pr.
I really not the user of readline and have zero motivation to do it in readline. The code in the readline shell so different that requres to additional redesign to make probes similar with ptk. So I leave the idea of adding it to readline.

Reply:
Well, this is more of a change to the API that about when the timings are reported. So if I understand this PR correctly, it breaks timings with readline.  This should be a simple one-line fix

Reply:
done